src/s3/destination.ts

```diff
@@ -128,62 +199,74 @@ export class S3Destination implements IDestination {
 
     return {
       connectorType: this.connectorType.asProfileConnectorType,
-      destinationConnectorProperties: this.buildDestinationConnectorProperties(),
+      destinationConnectorProperties:
+        this.buildDestinationConnectorProperties(),
     };
   }
 
   /**
    * see: https://docs.aws.amazon.com/appflow/latest/userguide/security_iam_id-based-policy-examples.html#security_iam_id-based-policy-examples-access-gdc
    * see: https://docs.aws.amazon.com/appflow/latest/userguide/security_iam_service-role-policies.html#access-gdc
    * @param flow
+   * @param database - the AWS Glue database
+   * @param tablePrefix - the prefix for the tables that will be created in the AWS Glue database
    * @returns a tuple consisting of a role for cataloguing with a specified policy
    */
-  private buildRoleAndPolicyForCatalog(flow: IFlow) {
+  private buildRoleAndPolicyForCatalog(
+    flow: IFlow,
+    database: IDatabase,
+    tablePrefix: string,
+  ) {
     const role = new Role(flow.stack, `${flow.node.id}GlueAccessRole`, {
       assumedBy: new ServicePrincipal('appflow.amazonaws.com'),
     });
 
-    const policy = new Policy(flow.stack, `${flow.node.id}GlueAccessRolePolicy`, {
-      roles: [role],
-      statements: [
-        new PolicyStatement({
-          effect: Effect.ALLOW,
-          actions: [
-            'glue:BatchCreatePartition',
-            'glue:CreatePartitionIndex',
-            'glue:DeleteDatabase',
-            'glue:GetTableVersions',
-            'glue:GetPartitions',
-            'glue:BatchDeletePartition',
-            'glue:DeleteTableVersion',
-            'glue:UpdateTable',
-            'glue:DeleteTable',
-            'glue:DeletePartitionIndex',
-            'glue:GetTableVersion',
-            'glue:CreatePartition',
-            'glue:UntagResource',
-            'glue:UpdatePartition',
-            'glue:TagResource',
-            'glue:UpdateDatabase',
-            'glue:CreateTable',
-            'glue:BatchUpdatePartition',
-            'glue:GetTables',
-            'glue:BatchGetPartition',
-            'glue:GetDatabases',
-            'glue:GetPartitionIndexes',
-            'glue:GetTable',
-            'glue:GetDatabase',
-            'glue:GetPartition',
-            'glue:CreateDatabase',
-            'glue:BatchDeleteTableVersion',
-            'glue:BatchDeleteTable',
-            'glue:DeletePartition',
-          ],
-          resources: ['*'],
```

Comment:
The only action I'm unsure about is `glue:UntagResource`, which operates on a bunch of different resources: https://docs.aws.amazon.com/service-authorization/latest/reference/list_awsglue.html#awsglue-connection

![Screenshot 2024-10-08 at 07 30 48](https://github.com/user-attachments/assets/8a5b7010-cb16-444a-936b-37534d244a57)

My guess is that it only deals with database and schemas, but I'm not positive.

Reply:
Please take a look [here](https://docs.aws.amazon.com/glue/latest/dg/monitor-tags.html). One can tag/untag databases so you're right. I don't think we need to break down this policy into smaller as the combination of actions and resources gives us *least privilege*

Reply:
My concern here is if we need to allow `glue:UntagResource` on `*` to retain good behavior. But, like you said, it might just need to work with catalog/database/table, so we could start there.

Just noting that if something breaks we might need a separate statement that allows Tag/Untag resource on a wider swath of resources.